kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java

```diff
@@ -159,15 +176,24 @@ public void startIfEnabled() {
 
                                 log.debug("Deleting item with ref {}. It's been successfully converted.", ref);
 
-                                repository.deleteById(inProgressItem.getId());
+                                // We just want to hard delete it here. No need to requery.
+                                repository.deleteById(item.getId());
                             } catch (Exception e) {
-                                // Too many attempts. Do not try again.
-                                if (inProgressItem.getAttempts() > maxAttemptsAllowed) {
-                                    inProgressItem.setStatus(FileConversionQueueItem.Status.INVALID);
-                                } else {
-                                    inProgressItem.setStatus(FileConversionQueueItem.Status.NOT_STARTED);
-                                }
-                                repository.save(inProgressItem);
+                                (new TransactionTemplate(transactionManager)).execute(new TransactionCallbackWithoutResult() {
```

Comment:
Can actually create the tx template with spring beans.

Reply:
not sure how to do this. @adrianfish ?